src/Http/Controllers/VoyagerSettingsController.php

```diff
@@ -89,7 +89,10 @@ public function update(Request $request)
                 'group'   => $setting->group,
             ]);
 
-            if ($content === null && isset($setting->value)) {
+            if ($content === null && $setting->type !== 'image') {
+                $content = '';
+            }
+            if ($content === null && $setting->type === 'image') {
```

Comment:
I don't really like the code style here.  It's difficult to read.

But beyond that, you've changed the logic by removing `isset()`.  Also, you've seemingly added an arbitrary check for the "image" type, with no explanation why that particular type is excluded.

Reply:
thanks for your review
i excluded image because image in update action will return with a null input but image may have a value in database if i remove this check the value in database will be removed

Reply:
That makes sense for images (and is what I expected your answer to be), but are you certain there are no other types that need to be excluded?

Reply:
```
<select name="type" class="form-control" required="required">
                            <option value="">{{ __('voyager.generic.choose_type') }}</option>
                            <option value="text">{{ __('voyager.form.type_textbox') }}</option>
                            <option value="text_area">{{ __('voyager.form.type_textarea') }}</option>
                            <option value="rich_text_box">{{ __('voyager.form.type_richtextbox') }}</option>
                            <option value="code_editor">{{ __('voyager.form.type_codeeditor') }}</option>
                            <option value="checkbox">{{ __('voyager.form.type_checkbox') }}</option>
                            <option value="radio_btn">{{ __('voyager.form.type_radiobutton') }}</option>
                            <option value="select_dropdown">{{ __('voyager.form.type_selectdropdown') }}</option>
                            <option value="file">{{ __('voyager.form.type_file') }}</option>
                            <option value="image">{{ __('voyager.form.type_image') }}</option>
 </select>
```


from code there is two types file and image